src/contracts/Protocol/Vault.sol

```diff
@@ -76,7 +76,8 @@ contract Vault is Ownable, OnlyCostController, OnlyApi, VaultEventsAndErrors {
         // Checks if METRIC is locked for a valid stage.
         if (uint8(stage) >= 5) revert InvalidStage();
         // Checks if there has not been a deposit yet
-        if (lockedMetric[questionId][stage][user].status != STATUS.UNINT) revert QuestionHasInvalidStatus();
+        // TODO remove amount!=0 possibly
+        if (lockedMetric[questionId][stage][user].status != STATUS.UNINT && amount != 0) revert QuestionHasInvalidStatus();
```

Comment:
I think this works for now - but I'm thinking we should change tests as well because a user shouldn't be creating a question and then publishing it

Reply:
Wouldn't introducing amount != 0 mean that we cannot set cost to 0? Also think published should be a separate state, otherwise they conflict (explained in slack)

Reply:
no it means you can't have a question that costs money to create and cost money to publish

Reply:
that was the intention but I updated this to not to do that